pkg/option/config.go

```diff
@@ -1039,6 +1039,17 @@ func getIPv4Enabled() bool {
 	return viper.GetBool(EnableIPv4Name)
 }
 
+func getIPv6Enabled() bool {
+	enableIPv6 := viper.GetBool(EnableIPv6Name)
+	if enableIPv6 {
+		if _, err := os.Stat("/proc/net/if_inet6"); os.IsNotExist(err) {
+			return false
```

Comment:
Can you add a info message stating that you are disabling IPv6 because there's no kernel module for IPv6?

Reply:
@aanm this happens in initEnv(), before we setup logging. I wonder how to handle such case,  shall I use fmt.Print

Reply:
Good point. My personal opinion - I would merge that PR, then make a separate PR (or multiple PRs) which will:

- create a `log` variable with `option` subsystem in `pkg/option`
- initialize logger in `daemon` before this module is called

Reply:
@aanm @tgraf what's your opinion?

Reply:
There are plenty of validation checks being made in the daemon like:

https://github.com/cilium/cilium/blob/889fd1d0c40a18873227776e883c85315c89866f/daemon/daemon.go#L1041-L1050

Made we should consider fatal the agent if the user tries enable the v6 flag but there is not kernel module loaded.

Reply:
At least in the K8s case, I believe that we can do a `fmt.Printf` and it will be saved in the K8s logs. 
I think that exiting is a valid suggestion since this is something the user specified that we cannot recover from.

Reply:
Disabling IPv6 without notifying the user is pretty bad. It's completely invisible the user on why that happened. I suggest we `Fatalf()` if IPv6 is enabled but not compiled in. IPv6 is disabled by default so the user has to opt into this and will expect it to work or cilium to fail.